DATA/production/configurations/2022/MayJunePilotBeam/apass1/selectSettings.sh

```diff
@@ -23,8 +23,85 @@ if [[ $RUNNUMBER -ge 519903 ]] && [[ $RUNNUMBER -le 520099 ]]; then
   COLLISIONCONTEXT=collisioncontext_Single_4b_2_2_2_noLR.root
 fi
 
+# LHC22f: 520143 - 520473
+if [[ $RUNNUMBER -ge 520143 ]] && [[ $RUNNUMBER -le 520473 ]]; then
+  # +30kA/+6kA
+  GRPMAG=o2sim_grp_b5p_128HB.root
+fi
+
+#       FILL 7920: 520143 - 520163
+if [[ $RUNNUMBER -ge 520143 ]] && [[ $RUNNUMBER -le 520163 ]]; then
+  COLLISIONCONTEXT=collisioncontext_Single_3b_2_2_2.root
+fi
+
+#       FILL 7921: 520175 - 520210
+if [[ $RUNNUMBER -ge 520175 ]] && [[ $RUNNUMBER -le 520210 ]]; then
+  COLLISIONCONTEXT=collisioncontext_Single_3b_2_2_2.root
+fi
+#       FILL 7923: 520259 - 520297
+if [[ $RUNNUMBER -ge 520259 ]] && [[ $RUNNUMBER -le 520297 ]]; then
+  COLLISIONCONTEXT=collisioncontext_Single_12b_8_8_8_2018.root
+fi
+#       FILL 7960: 520471 - 520477
+if [[ $RUNNUMBER -ge 520471 ]] && [[ $RUNNUMBER -le 520477 ]]; then
+  COLLISIONCONTEXT=collisioncontext_25ns_75b_62_32_62_12bpi_9inj.root
+fi
+
+# LHC22g: 520474 - 520477
+if [[ $RUNNUMBER -ge 520474 ]] && [[ $RUNNUMBER -le 520477 ]]; then
+  # +12kA/+6kA
+  GRPMAG=o2sim_grp_b2p_128HB.root
+fi
+
+
+# LHC22h: 520495 - 520509
+if [[ $RUNNUMBER -ge 520495 ]] && [[ $RUNNUMBER -le 520509 ]]; then
+  # 0kA/0kA
+  GRPMAG=o2sim_grp_b0_128HB.root
+fi
+
+#       FILL 7963: 520495 - 520509
+if [[ $RUNNUMBER -ge 520495 ]] && [[ $RUNNUMBER -le 520509 ]]; then
+  COLLISIONCONTEXT=collisioncontext_25ns_75b_62_32_62_12bpi_9inj.root
+fi
+
+# LHC22i: 520529 - 520542
+if [[ $RUNNUMBER -ge 520529 ]] && [[ $RUNNUMBER -le 520542 ]]; then
+  # -12kA/-6kA
+  GRPMAG=o2sim_grp_b2m_128HB.root
+fi
+
+#       FILL 7965: 520529 - 520529
+if [[ $RUNNUMBER -ge 520529 ]] && [[ $RUNNUMBER -le 520529 ]]; then
+  COLLISIONCONTEXT=collisioncontext_25ns_75b_62_32_62_12bpi_9inj.root
+fi
+#       FILL 7966: 520541 - 520560
+if [[ $RUNNUMBER -ge 520541 ]] && [[ $RUNNUMBER -le 520560 ]]; then
+  COLLISIONCONTEXT=collisioncontext_25ns_75b_62_32_62_12bpi_9inj.root
+fi
+
+# LHC22j: 520543 - ...
+if [[ $RUNNUMBER -ge 520543 ]] && [[ $RUNNUMBER -le 999999 ]]; then
+  # -30kA/-6kA
+  GRPMAG=o2sim_grp_b5m_128HB.root
+fi
+
+#       FILL 7967: 520567 - 520581
+if [[ $RUNNUMBER -ge 520567 ]] && [[ $RUNNUMBER -le 520581 ]]; then
+  COLLISIONCONTEXT=collisioncontext_525ns_72b_64_48_16_8bpi_9inj_800ns_bs525ns.root
+fi
+#       FILL 7969: 520703 - 520714
+if [[ $RUNNUMBER -ge 520703 ]] && [[ $RUNNUMBER -le 520714 ]]; then
+  COLLISIONCONTEXT=collisioncontext_25ns_339b_326_206_266_12bpi_31inj.root
+fi
+#       FILL 7978: 520884 - 520904
+if [[ $RUNNUMBER -ge 520884 ]] && [[ $RUNNUMBER -le 520904 ]]; then
+  COLLISIONCONTEXT=collisioncontext_25ns_315b_302_237_240_48bpi_11inj.root
+fi
+
+
 # B field update
-if [[ $RUNNUMBER -ge 519041 ]]; then
+if [[ $RUNNUMBER -ge 999999 ]]; then
```

Comment:
I don't understand this change, sorry. 
Also, we need to reflect the correct B field for QC.

Reply:
What do you mean "for QC"?

Reply:
QC might need to have the GRP in the local directory. Using a default would lead to wrong results. And there was anyway a default, so I don’t understand the change - unless i am overlooking something.

Anyway, due to QC, we need to keep this file updated with the running conditions, i fear.

Reply:
If I put the last run of LHC22j, tomorrow it will be already obsolete before the PR is merged.

Reply:
Then probably you wanted to have "-le" 999999? but then it will override what you have in line 86. You then need some `else`.
Sorry if I am not seeing the obvious. 
"-ge 999999" will never be true (for now at least).

Reply:
I removed that part.
My problem was that I didn't understand the role of 

if [[ $RUNNUMBER -ge 519041 ]]; then
  GRPMAG=o2sim_grp_b5p_128HB.root
fi

in the previous version

Reply:
It was leaving as default the latest B that had been set, overriding the default at the top of the script. The latest should always become the default.